src/database/migrations/0031-create-sources.sql

```diff
@@ -0,0 +1,33 @@
+CREATE TYPE source_type AS ENUM ('data_provider', 'funder', 'organization', 'system');
+
+CREATE TABLE sources (
+  id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  source_type source_type NOT NULL,
+  label VARCHAR NOT NULL,
+  organization_id INTEGER REFERENCES organizations(id),
+  funder_id INTEGER REFERENCES funders(id),
+  data_provider_id INTEGER REFERENCES data_providers(id),
+  created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
+  UNIQUE(organization_id, funder_id, data_provider_id),
+  CONSTRAINT chk_proper_fk CHECK (
```

Comment:
it would probably be cleaner to actually define a separate constraint for each type with a proper constraint name.

One benefit of this is that if a type is ever added or removed we simply add or remove a relevant constraint, rather than having to re-write the entire constraint.  Another is that the constraint names would be more meaningful in errors.